client/components/QueryListDropdown/QueryListDropdown.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Popover, InputGroup, PopoverPosition } from '@blueprintjs/core';
+import { IQueryListRendererProps, ItemRenderer, QueryList } from '@blueprintjs/select';
+
+require('./queryListDropdown.scss');
+
+type Props<Item> = {
+	children: React.ReactNode;
+	items: Item[];
+	itemPredicate: (query: string, item: Item) => boolean;
+	onItemSelect: (item: Item) => unknown;
+	position?: PopoverPosition;
+	searchPlaceholder: string;
+	emptyListPlaceholder: React.ReactNode;
+	itemRenderer: ItemRenderer<Item>;
+};
+
+const QueryListDropdown = <Item extends {}>(props: Props<Item>) => {
+	const {
+		children,
+		emptyListPlaceholder,
+		itemPredicate,
+		items,
+		onItemSelect,
+		position = 'bottom-right',
+		itemRenderer,
+		searchPlaceholder,
+	} = props;
+
+	const renderPopoverContent = (qlProps: IQueryListRendererProps<Item>) => {
+		const { handleKeyDown, handleKeyUp, handleQueryChange, itemList } = qlProps;
+		return (
+			// eslint-disable-next-line jsx-a11y/no-static-element-interactions
+			<div className="search-wrapper" onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+				<InputGroup
+					inputRef={(input) => input && setTimeout(() => input.focus(), 0)}
```

Comment:
It's somewhat surprising that `InputGroup` doesn't have an `autofocus` attribute.